std/file.d

```diff
@@ -446,6 +450,12 @@ Throws: $(D FileException) on error.
 void write(R)(R name, const void[] buffer)
     if ((isInputRange!R && isSomeChar!(ElementEncodingType!R) || isSomeString!R) &&
         !isConvertibleToString!R)
+in
+{
+    version(Posix) assert(buffer.length <= 0x7ffff000);
+    version(Win64) assert(buffer.length <= uint.max);
```

Comment:
Looks like  a CT constraints can be added c`version(...) ... else version(...) else static assert(0,  "std.file.write constraints is not defined for this platform.");`